lua/tardis/sh_menuoptions.lua

```diff
@@ -29,26 +29,107 @@ if CLIENT then
 			for k,v in pairs(TARDIS:GetInteriors()) do
 				if selectedinterior==v.ID then
-					comboBox:ChooseOption(v.OptionID)
-					comboBox:SetText(v.Name)
+					interior_combobox:ChooseOption(v.OptionID)
+					interior_combobox:SetText(v.Name)
 				end
 			end
-			comboBox.OnSelect = function(panel,index,value,data)
+			interior_combobox.OnSelect = function(panel,index,value,data)
 				TARDIS:SetSetting("interior",data,true)
 				LocalPlayer():ChatPrint("TARDIS interior changed. Respawn or repair the TARDIS for changes to apply.")
 			end
-			panel:AddItem(comboBox)
+			panel:AddItem(interior_combobox)
 
+			local visual_gui_toggle = vgui.Create("DCheckBoxLabel")
+			visual_gui_toggle:SetText("Enable new visual GUI")
+			visual_gui_toggle:SetValue(TARDIS:GetSetting("visual_gui_enabled"))
+			function visual_gui_toggle:OnChange(val)
+				TARDIS:SetSetting("visual_gui_enabled", val)
+			end
+			panel:AddItem(visual_gui_toggle)
+
+			local visual_gui_big_toggle = vgui.Create("DCheckBoxLabel")
+			visual_gui_big_toggle:SetText("Visual GUI big popup")
+			visual_gui_big_toggle:SetValue(TARDIS:GetSetting("visual_gui_bigpopup"))
+			function visual_gui_big_toggle:OnChange(val)
+				TARDIS:SetSetting("visual_gui_bigpopup", val)
+			end
+			panel:AddItem(visual_gui_big_toggle)
+
+			local DLabel3 = vgui.Create( "DLabel" )
+			DLabel3:SetText("Number of visual GUI button rows on the screen:")
+			panel:AddItem(DLabel3)
+			local visual_gui_screen_numrows = vgui.Create("DNumSlider")
+			visual_gui_screen_numrows:SetMinMax(2, 10)
+			visual_gui_screen_numrows:SetDecimals(0)
+			visual_gui_screen_numrows:SetValue(TARDIS:GetSetting("visual_gui_screen_numrows"))
+			function visual_gui_screen_numrows:OnValueChanged(val)
+				val = math.floor(val)
+				visual_gui_screen_numrows:SetValue(val)
+				TARDIS:SetSetting("visual_gui_screen_numrows", val)
+			end
+			panel:AddItem(visual_gui_screen_numrows)
+
+			local DLabel4 = vgui.Create( "DLabel" )
+			DLabel4:SetText("Number of visual GUI button rows in the popup:")
+			panel:AddItem(DLabel4)
+			local visual_gui_popup_numrows = vgui.Create("DNumSlider")
+			visual_gui_popup_numrows:SetMinMax(2, 10)
+			visual_gui_popup_numrows:SetDecimals(0)
+			visual_gui_popup_numrows:SetValue(TARDIS:GetSetting("visual_gui_popup_numrows"))
+			function visual_gui_popup_numrows:OnValueChanged(val)
+				val = math.floor(val)
+				visual_gui_popup_numrows:SetValue(val)
+				TARDIS:SetSetting("visual_gui_popup_numrows", val)
+			end
+			panel:AddItem(visual_gui_popup_numrows)
+
+			local DLabel4 = vgui.Create( "DLabel" )
+			DLabel4:SetText("TARDIS Visual GUI Theme:")
+			panel:AddItem(DLabel4)
+
+			local visual_gui_theme = vgui.Create("DComboBox")
+			visual_gui_theme:SetText("Visual GUI Theme")
+			local themes = {}
+			local theme_basefolder = TARDIS.visualgui_theme_basefolder
+			local files, folders = file.Find(theme_basefolder.."*", "GAME", "nameasc")
+			for k,folder in pairs(folders) do
```

Comment:
Themes should have a "base" concept to reduce duplication, as we can see with the default / vortex everything is copied apart from the background which isn't great at all. I would recommend building a manifest of all the files for each folder, starting from the furthest down one that has no base at all, and then overriding any existing files as you go back up to the top level, and saving that in a global variable somewhere which can be accessed to find appropriate textures without duplicating them - much like how the AddInterior code works. Happy to help you out on this bit if you need help.

Reply:
Replied in Discord

Reply:
Happy with this for now, will raise an issue to track

Reply:
Closing this conversation with issue moved to: 
https://github.com/MattJeanes/TARDIS/issues/360#issuecomment-784166918